library/core/src/alloc/layout.rs

```diff
@@ -372,38 +372,46 @@ impl Layout {
     /// padding is inserted, the alignment of `next` is irrelevant,
     /// and is not incorporated *at all* into the resulting layout.
     ///
-    /// On arithmetic overflow, returns `LayoutErr`.
+    /// On arithmetic overflow, returns `LayoutError`.
     #[unstable(feature = "alloc_layout_extra", issue = "55724")]
     #[inline]
-    pub fn extend_packed(&self, next: Self) -> Result<Self, LayoutErr> {
-        let new_size = self.size().checked_add(next.size()).ok_or(LayoutErr { private: () })?;
+    pub fn extend_packed(&self, next: Self) -> Result<Self, LayoutError> {
+        let new_size = self.size().checked_add(next.size()).ok_or(LayoutError { private: () })?;
         Layout::from_size_align(new_size, self.align())
     }
 
     /// Creates a layout describing the record for a `[T; n]`.
     ///
-    /// On arithmetic overflow, returns `LayoutErr`.
+    /// On arithmetic overflow, returns `LayoutError`.
     #[stable(feature = "alloc_layout_manipulation", since = "1.44.0")]
     #[inline]
-    pub fn array<T>(n: usize) -> Result<Self, LayoutErr> {
+    pub fn array<T>(n: usize) -> Result<Self, LayoutError> {
         let (layout, offset) = Layout::new::<T>().repeat(n)?;
         debug_assert_eq!(offset, mem::size_of::<T>());
         Ok(layout.pad_to_align())
     }
 }
 
+#[stable(feature = "alloc_layout", since = "1.28.0")]
+#[rustc_deprecated(
+    since = "1.51.0",
+    reason = "use LayoutError instead",
```

Comment:
I think this should probably say something like, "name doesn't follow std convention." Currently, it's redundant with `suggestion` below.

Reply:
I chose this as the reason because it is the string shown in the warning message:
```
error: use of type alias `core::alloc::LayoutErr` that will be deprecated in future version 1.51.0: use LayoutError instead
 --> library/alloc/src/raw_vec.rs:4:5
  |
4 | use core::alloc::LayoutErr;
  |     ^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `-D deprecated-in-future` implied by `-D warnings`

error: use of type alias `core::alloc::LayoutErr` that will be deprecated in future version 1.51.0: use LayoutError instead
   --> library/alloc/src/raw_vec.rs:474:32
    |
474 |     new_layout: Result<Layout, LayoutErr>,
    |                                ^^^^^^^^^
```

Perhaps something along the lines of "Name does not follow std convention, use LayoutError" would work?

Reply:
I guess so, but it looking a bug in the error reporting to me? Your suggestion sounds fine for now. Thanks. :)

Reply:
Fixed in the latest commit :)